packages/server/src/user/user.service.ts

```diff
@@ -0,0 +1,164 @@
+import { Injectable } from '@nestjs/common';
+import { Prisma, USER } from '@prisma/client';
+import { PrismaService } from '../prisma/prisma.service';
+
+@Injectable()
+export class UserService {
+  constructor(private prisma: PrismaService) {}
+
+  // TO BE REMOVED
+  // temp user list for testing purposes
+  private readonly users: USER[] = [
+    {
+      id: 'a31247c5-2632-45e8-b7e2-746ba5a2d1e8',
+      project_id: 'project-001',
+      username: 'user0',
+      email: 'some.user@mail.com',
+      password: 'some_user_password',
+      role: 0,
+      created_at: null,
+      updated_at: null,
+      deleted_at: null,
+      reset_code: 'some_reset_code',
+      reset_code_expires_at: null
+    },
+    {
+      id: '94ea3787-16e0-4a5c-b9a5-d345f4fa82dd',
+      project_id: 'project-001',
+      username: 'user1',
+      email: 'another.user@mail.com',
+      password: 'another_user_password',
+      role: 0,
+      created_at: null,
+      updated_at: null,
+      deleted_at: null,
+      reset_code: null,
+      reset_code_expires_at: null
+    },
+    {
+      id: 'e929ebd5-a899-4df2-a94c-0f024de39cbd',
+      project_id: 'project-002',
+      username: 'admin0',
+      email: 'some.admin@mail.com',
+      password: 'some_admin_password',
+      role: 1,
+      created_at: null,
+      updated_at: null,
+      deleted_at: null,
+      reset_code: null,
+      reset_code_expires_at: null
+    }
+  ];
+
+  /**
+   * Create a new record of user in the database
+   *
+   * @param data object that should contains `project_id: string`, `username?: string`, `email: string` and `password: string` in plain text
+   * @returns User object
+   */
+  async createUser(data: Prisma.USERCreateInput): Promise<USER> {
+    // TO BE ADDED
+
+    // TO BE REMOVED
+    // create and return a dummy user for testing
+    const user = {
+      id: '259aab6e-ac66-4ad0-b2ca-6671336d67bf',
+      project_id: data.project_id,
+      username: data.username,
+      email: data.username,
+      password: data.password,
+      role: 0,
+      created_at: null,
+      updated_at: null,
+      deleted_at: null,
+      reset_code: null,
+      reset_code_expires_at: null
+    };
+
+    this.users.push(user);
+
+    return user;
+  }
+
+  /**
+   * Find unique user for a project using email.
+   *
+   * @param params object includes: `project_id`, `username`, and `password` in plain text
+   * @returns User object
+   */
+  async findUserByUsername(params: { project_id; username; password }): Promise<USER> {
```

Comment:
Do we want to have to always have the password when searching for a user? It seems like the code base would then have to have the password anytime any user information is needed.

Reply:
It should be the function for login. The name could be confusing, so maybe we can change that to loginByUsername and loginByEmail?

Reply:
Yea I think if its going to be the login function maybe renaming it to make that more clear. In the future we will probably want to be able to find users not in a login context (like checking to see if the username is available or not) where we will not necessarily have the password on hand